airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -934,13 +934,22 @@ def var_print(var_name):
         else:
             return "Format exception for {var}: ".format(var=var_name)
 
-    cmpt = table_input.split(':')
+    if table_input.count('.') + table_input.count(':') > 3:
```

Comment:
A little confused by this logic. Aren't all of the following valid:

proj.dataset.table
proj:dataset.table
some:proj:dataset.table
some:other:proj:datset.table

The last one seems to break under your logic:

```
>>> table_input = 'some:other:proj:datset.table'
>>> table_input.count('.') + table_input.count(':')
4
```